openid-4-verifiable-presentations-1_0.md

```diff
@@ -595,11 +601,265 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+
+`namespace`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED if the Credential Format is based on mdoc format defined in ISO 18013-5; MUST NOT be present otherwise. 
+The value MUST be a string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+`value`:
+: OPTIONAL. A string, integer or boolean value that specifies the expected value of the claim. If the
+`value` property is present, the Wallet MUST return the claim only if the type and value
+of the claim match the type and value specified in the query. This property MUST NOT be present if the `values` property is present.
+
+`values`:
+: OPTIONAL. An array of strings, integers or boolean values that specifies the expected values of the claim.
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array. This property MUST NOT be present if the `value` property is present.
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The same basic logic applies for selecting claims and for selecting credentials, as detailed in the following.
+
+#### Selecting Claims
+
+The following rules apply for selecting claims via `claims` and `claim_sets`:
+
+- If `claims` is not provided, the Verifier requests all claims existing
+  in the Credential.
+- If `claims` is provided, but `claim_sets` is not provided,
+  the Verifier requests all claims listed in `claims`.
+- Otherwise, the Verifier requests one combination of the claims listed in
+  `claim_sets`, with optional claims marked by the postfix `?`.
+
+If the Wallet cannot deliver all non-optional claims requested by the Verifier according to these rules, it MUST NOT
+return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `credentials` and `credential_sets`:
+
+- If `credential_sets` is not provided, the Verifier requests all
+  Credentials in `credentials` to be returned.
+- Otherwise, the Verifier requests one combination of the Credentials
+  listed in `credential_sets`, with optional credentials marked by the postfix `?`.
+
+Credentials not matching the respective constraints expressed within
+`credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the Verifier according to these rules, it MUST NOT
+return any credential(s).
+
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
+
+### Format `vc+sd-jwt` {#format_vc_sd_jwt}
```

Comment:
we should use media types for all formats, or define our own formats

Reply:
```suggestion
### IETF SD-JWT VC {#format_vc_sd_jwt}
```

+1

Reply:
I'd recommend against the media type here, but I would make it clear how to support whichever media types correspond to the structure of the claimsets that are desired by implementers of the spec, and that might require more than just listing the media type at this point.